pkg/contract/contract.go

```diff
@@ -1,225 +1,121 @@
-		return nil, fmt.Errorf("%w: could not match the contract template regex", ErrInvalidContractText)
+		return fmt.Errorf("%w: could not match the contract template regex", ErrInvalidContractText)
 	}
 	matches := matchResult[1:]
 
-	if len(matches) != len(c.TemplateAttributes) {
-		return nil, fmt.Errorf("%w: amount of template attributes does not match the amount of params: found: %d, expected %d", ErrInvalidContractText, len(matches), len(c.TemplateAttributes))
+	if len(matches) != len(sc.Template.TemplateAttributes) {
+		return fmt.Errorf("%w: amount of template attributes does not match the amount of Params: found: %d, expected %d", ErrInvalidContractText, len(matches), len(sc.Template.TemplateAttributes))
 	}
 
-	result := make(map[string]string, len(matches))
-
-	for i, m := range matches {
-		result[c.TemplateAttributes[i]] = string(m)
-	}
-
-	return result, nil
-}
-
-func parseTime(timeStr string, _ Language) (*time.Time, error) {
-	amsterdamLocation, _ := time.LoadLocation("Europe/Amsterdam")
-	parsedTime, err := monday.ParseInLocation(timeLayout, timeStr, amsterdamLocation, monday.LocaleNlNL)
-	if err != nil {
-		return nil, fmt.Errorf("invalid time string [%v]: %w", timeStr, err)
+	sc.Params = make(map[string]string, len(matches))
+	for idx, match := range matches {
+		sc.Params[sc.Template.TemplateAttributes[idx]] = string(match)
 	}
-	return &parsedTime, nil
+	return nil
 }
 
-func (c Template) ValidateTimeFrame(params map[string]string) (bool, error) {
+// Verify verifies the params with the template
+func (sc Contract) Verify() error {
 	var (
 		err                      error
 		ok                       bool
 		validFrom, validTo       *time.Time
 		validFromStr, validToStr string
 	)
 
-	if validFromStr, ok = params["valid_from"]; !ok {
-		return false, fmt.Errorf("%w: value for [valid_from] is missing", ErrInvalidContractText)
+	if validFromStr, ok = sc.Params[ValidFromAttr]; !ok {
+		return fmt.Errorf("%w: value for ["+ValidFromAttr+"] is missing", ErrInvalidContractText)
 	}
 
-	validFrom, err = parseTime(validFromStr, c.Language)
+	validFrom, err = parseTime(validFromStr, sc.Template.Language)
 	if err != nil {
-		return false, fmt.Errorf("%w: unable to parse [valid_from]: %s", ErrInvalidContractText, err)
+		return fmt.Errorf("%w: unable to parse ["+ValidFromAttr+"]: %s", ErrInvalidContractText, err)
 
 	}
 
-	if validToStr, ok = params["valid_to"]; !ok {
-		return false, fmt.Errorf("%w: value for [valid_to] is missing", ErrInvalidContractText)
+	if validToStr, ok = sc.Params[ValidToAttr]; !ok {
+		return fmt.Errorf("%w: value for ["+ValidToAttr+"] is missing", ErrInvalidContractText)
 	}
 
-	validTo, err = parseTime(validToStr, c.Language)
+	validTo, err = parseTime(validToStr, sc.Template.Language)
 	if err != nil {
-		return false, fmt.Errorf("%w: unable to parse [valid_to]: %s", ErrInvalidContractText, err)
+		return fmt.Errorf("%w: unable to parse ["+ValidToAttr+"]: %s", ErrInvalidContractText, err)
```

Comment:
mixed concatenation and use of formatter. I think Rein suggested to make the entire error a var instead of formatting.

```
var ErrInvalidPeriod = fmt.Errorf("%w: [valid_from] must be after [valid_to]", ErrInvalidContractText)
```